src/shared-components/icon/index.tsx

```diff
@@ -43,7 +47,8 @@ Icon.propTypes = {
   displayInline: PropTypes.bool,
   fill: PropTypes.string,
   height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-  IconComponent: PropTypes.elementType.isRequired,
+  // @ts-expect-error -- TypeScript type is too specific for PropTypes types
+  IconComponent: PropTypes.element.isRequired,
```

Comment:
Could not figure this out, but since it is only ever used internally ignoring it doesn't matter for consumer app usage.